test/cufinufft1d2_test.cu

```diff
@@ -0,0 +1,187 @@
+#include <iostream>
+#include <iomanip>
+#include <math.h>
+#include <helper_cuda.h>
+#include <complex>
+
+#include <cufinufft_eitherprec.h>
+
+#include <profile.h>
+#include "../contrib/utils.h"
+
+using namespace std;
+
+int main(int argc, char* argv[])
+{
+	int N1, M;
+	if (argc<3) {
+		fprintf(stderr,
+			"Usage: cufinufft2d2_test method N1 [M [tol]]\n"
+			"Arguments:\n"
+			"  method: One of\n"
+			"    1: nupts driven\n"
+			"  N1: The size of the 1D array.\n"
+			"  M: The number of non-uniform points (default N1).\n"
+			"  tol: NUFFT tolerance (default 1e-6).\n");
+		return 1;
+	}
+	double w;
+	int method;
+	sscanf(argv[1],"%d",&method);
+	sscanf(argv[2],"%lf",&w); N1 = (int)w;  // so can read 1e6 right!
+	M = N1;// let density always be 1
+	if(argc>3){
+		sscanf(argv[3],"%lf",&w); M  = (int)w;  // so can read 1e6 right!
+	}
+
+	FLT tol=1e-6;
+	if(argc>4){
+		sscanf(argv[4],"%lf",&w); tol  = (FLT)w;  // so can read 1e6 right!
+	}
+	int iflag=1;
+
+
+	cout<<scientific<<setprecision(3);
+	int ier;
+
+
+	FLT *x;
+	CPX *c, *fk;
+	cudaMallocHost(&x, M*sizeof(FLT));
+	cudaMallocHost(&c, M*sizeof(CPX));
+	cudaMallocHost(&fk,N1*sizeof(CPX));
+
+	FLT *d_x;
+	CUCPX *d_c, *d_fk;
+	checkCudaErrors(cudaMalloc(&d_x,M*sizeof(FLT)));
+	checkCudaErrors(cudaMalloc(&d_c,M*sizeof(CUCPX)));
+	checkCudaErrors(cudaMalloc(&d_fk,N1*sizeof(CUCPX)));
+	// Making data
+	for (int i = 0; i < M; i++) {
+		x[i] = M_PI*randm11();// x in [-pi,pi)
+	}
+	for(int i=0; i<N1; i++){
+		fk[i].real(1.0);
```

Comment:
`1.0` → `randm11()`?

Reply:
Yes, sure. I updated it and also other type 2 tests in d8e24dd.

Reply:
Thanks.